### DISCUSSION TITLE
How to set a fixed rate for overall project, independent of tracked time

### DISCUSSION THREAD

**Discussion by maurice-kpw** [kimai/kimai#3386] [Questions & Answers] [answered]
Hello!

Is it possible to set a fixed rate for a project which has a fixed price while also being able to track the times / activities done?
When I try to set a fixed amount for a project and track time on different days for different activities (or the same activity for that matter) it shows up on the invoice multiple times.

day 1 | project x | description of what has been done | fixed rate
day 2 | project x | description of something different that has been done | fixed rate
sum: 2 x fixed rate

I want a list of descriptions that have been done for this project while being able to only have that fixed amount per project show up on the invoice 1 time as this is the price for the project the customer and we have agreed on. Is this possible? 
Or is this the feature that ~~has been removed~~ (Edit: It has not been removed. Sorry, I misread that.) as discussed here: https://github.com/kevinpapst/kimai2/discussions/3127

If that is the case, would that be possible after purchasing the expenses-plugin? 

I hope you can understand what we are trying to achieve here. :) 
If not let me know and I'll try to elaborate further.

Regards,
– Maurice

**Comment by kevinpapst** [comment, marked as answer]
Hi Maurice,

you misunderstand the use of fixed rates ... that's why I wanted to remove them, but anyway.

Fixed rates mean: no matter how long a time record is, it will get this rate. Can be useful for certain support contracts where e.g. each customer calls costs 10€. So everytime you log a "customer call" you will have 10€ more on your invoice.

What you want to achieve can be done by setting the project to non-billable, which then turns all new times into non-billable times (you can manually or bulk-change the old entries to be non-billable as well).
Now you can record whatever you want, it will not show up on your invoice.

The final invoice sum can then easily be generated e.g. by using the expense plugin and adding one item with the full project amount (what you called the "fixes price of the project").

So you have the entire working history in Kimai as non-billable items and the project cost as expense.
On top you can still validate your work by setting an "hour budget" to the project, which you can track in the project detail screen or in other reports.

**Reply by maurice-kpw** [reply]
Hey Kevin,

thanks for getting back to me. I guess we'll use the described workflow with the project being a fixed expense :) 

Thanks!
– Maurice

**Reply by coleshirley** [reply]
Throwing out an additional step that helps with internal project price tracking:

If you configure the project "Prices" section so that all users have a `0` as the price but leave the internal price set to the default value then all user time entries will not add to the billable amount of the project but will increase the internal price amount like so:

Before:
<img width="513" alt="image" src="https://github.com/kimai/kimai/assets/84236864/dc9881f1-fb80-4961-8917-c9727d7bda97">

After:
<img width="514" alt="image" src="https://github.com/kimai/kimai/assets/84236864/78850036-5579-419b-9338-f421141f9137">

Also make sure this setting is activated in system settings:
<img width="385" alt="image" src="https://github.com/kimai/kimai/assets/84236864/a34b2c6d-6adb-4e30-a3e4-759d9c9168d3">
